cfitsio: 3.21 -> 3.41

ndowens opened the pull request:
###### Motivation for this change
Update

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

vbgl commented:
`cfitsio` no longer builds on darwin: “ld: symbol(s) not found for architecture i386”. The full error can be read in the travis log.

matthiasbeyer reviewed (changes requested):
Commit message is IMO not appropriate.

Could you split this up into two commits: one for the package update and one for the platform change? Also - please add a short note in the second commit why the platform change is necessary.

Commit:
cfitsio: 3.21 -> 3.41

cfitsio: Changed platforms to Linux, due to failing to build on Darwin

cfitsio: Fixed missing semicolon

cfitsio: Removed undefined variable

ndowens commented:
@matthiasbeyer There we go, had to reset to old commit and redo it, which closed PR, but got it fixed now :)
